static/app/components/dropdownButtonV2.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import styled from '@emotion/styled';
+
+import Button from 'sentry/components/button';
+import {IconChevron} from 'sentry/icons';
+import space from 'sentry/styles/space';
+
+export type DropdownButtonProps = Omit<
+  React.ComponentProps<typeof Button>,
+  'type' | 'priority'
+> & {
+  /**
+   * The fixed prefix text to show in the button eg: 'Sort By'
+   */
+  prefix?: React.ReactNode;
+  /**
+   * Whether or not the button should render as open
+   */
+  isOpen?: boolean;
+  /**
+   * Should a chevron icon be shown?
+   */
+  showChevron?: boolean;
+  /**
+   * Button color
+   */
+  priority?: 'default' | 'primary' | 'form';
+  /**
+   * Forward a ref to the button's root
+   */
+  forwardedRef?: React.Ref<typeof Button>;
+};
+
+const DropdownButton = ({
+  children,
+  forwardedRef,
+  prefix,
+  isOpen = false,
+  showChevron = false,
+  disabled = false,
+  priority = 'form',
+  ...props
+}: DropdownButtonProps) => {
```

Comment:
If you're going to use a `const` function here, might as well drop the braces so it just becomes 

```tsx
}: DropdownButtonProps) => (
	  <StyledButton
		    {...props}
		    type="button"
		    disabled={disabled}
		    priority={priority}
		    isOpen={isOpen}
		    ref={forwardedRef}
	  >
			....
		</StyledButton>
);

```